### PR TITLE
[Booking_Service] added security

### PR DESCRIPTION
**Pull request by BohdanTrue**
- necessary dependencies
- Customer & Role entities/services/repositories
- SecurityConfig
- JwtAuthenticationFilter
- MapperConfig
- CustomerMapper
- Authentication Controller&Service

### CODE REVIEW

File: src/main/java/com/application/bookingservice/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableMethodSecurity
+@RequiredArgsConstructor
+public class SecurityConfig {
```
Comment (fedorovychh) [inline src/main/java/com/application/bookingservice/config/SecurityConfig.java:23]: move SecurityConfig to Security (Layer - Security)

File: src/main/java/com/application/bookingservice/dto/customer/CustomerLoginResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package com.application.bookingservice.dto.customer;
+
+public record CustomerLoginResponseDto(String token) {
+}
```
Comment (fedorovychh) [inline src/main/java/com/application/bookingservice/dto/customer/CustomerLoginResponseDto.java:4]: if we use classes, use classes everywhere

File: src/main/java/com/application/bookingservice/model/Customer.java [hunk 3]
```diff
@@ -1,4 +1,87 @@
...
+    private Long id;
+    @Column(nullable = false, unique = true)
+    private String email;
+    @Column(nullable = false)
+    private String firstName;
+    @Column(nullable = false)
+    private String lastName;
+    @Column(nullable = false)
+    private String password;
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (fedorovychh) [inline src/main/java/com/application/bookingservice/model/Customer.java:41]: Eager we don't use - BAD PRACTICE

**Review by fedorovychh — APPROVED** [review]
Check mistakes please

**Review by Alexander1iubchenko — APPROVED** [review]

**Pull request merged**